driver/wnbd.rc

```diff
@@ -0,0 +1,58 @@
+/*-
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (C) 2019-2020 SUSE LLC
+ *
+ * This library is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the
+ *
+ *  Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor,
+ *  Boston, MA 02110-1301 USA
+ *
+ */
+
+#include <windows.h>
+#include <ntverp.h>
+#include "..\include\version.h"
+
+#define VER_INTERNALNAME_STR        "wnbd.sys"
+#define VER_ORIGINALFILENAME_STR    "wnbd.sys"
+
+#undef VER_PRODUCTVERSION
+#define VER_PRODUCTVERSION          WNBD_VERSION_MAJOR,WNBD_VERSION_MINOR,WNBD_VERSION_PATCH,WNBD_COMMIT_COUNT
+
+#undef VER_PRODUCTVERSION_STR
+#define VER_PRODUCTVERSION_STR      WNBD_VERSION_STR
+
+#define VER_FILEVERSION             WNBD_VERSION_MAJOR,WNBD_VERSION_MINOR,WNBD_VERSION_PATCH,WNBD_COMMIT_COUNT
+#define VER_FILEVERSION_STR         WNBD_VERSION_STR_MS
+
+#undef VER_COMPANYNAME_STR
+#define VER_COMPANYNAME_STR        "SUSE LLC"
```

Comment:
All branding should go in variables ideally

Reply:
Through something like an include/vendor.h, or did you have something else in mind?

Reply:
A common way to do this is to directly assign #DEFINE from project property variables at build time, without generating header files, see for example:

https://github.com/cloudbase/cloudbase-init-installer/blob/master/BuildAutomation/BuildCloudbaseInitSetup.ps1#L154

Those project properties are defined in the project's property pages.

Reply:
Update: please disregard the last comment, beside automating the build we also want to ensure that people can just open the solution in VS and hit build, so having _version.h_ and _vendor.h_ files is the easiest approach rather than using variables.

What about adding a vendor.h file to the project with empty strings and a #warning to let people know they have to set those #defines?

Reply:
How about a vendor.h that uses a conditional (set by DefineConstants) to determine which vendor properties to apply? We can default to upstream (Cloudbase) strings, and easily allow extension by other vendors.

I've got an additional commit which shows this, and seems to work here. I'll push it shortly.

Reply:
What about leaving a generic "WNBD" for all three strings? The idea would be to overwrite Vendor.h during the build, this way we don't have to include all the branding info in the project itself.

Reply:
Ok, we can go that route for now.